### PR TITLE
Add instructions for M4 demo

### PR DESCRIPTION
**Pull request by davidmogar**
This PR adds resources and information for M4 demo. It uses kustomize to be able to run it in a dev environment (with CRC and Openshift Pipelines) and in staging (with a cluster bootstrapped with infra-deployments).

### CODE REVIEW

File: demos/m4/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+$ kubectl apply -k overlays/<environment>
+```
+
+## What to expect
+
+After running the commands above, you should expect to find:
+
+* a Build PipelineRun for the component `m4-component`;
+* a Release created in the `demo` namespace;
+* a Release PipelineRun created in the `tenant` namespace.
```
Comment (scoheb) [inline demos/m4/README.md:28]: this should be **managed** workspace

File: demos/m4/base/namespaces.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: demo
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: tenant
```
Comment (scoheb) [inline demos/m4/base/namespaces.yaml:9]: this should be **managed** workspace

File: demos/m4/base/release_links.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleaseLink
+metadata:
+  name: demo
+  namespace: demo
+spec:
+  displayName: Users's ReleaseLink
+  application: m4-app
+  target: tenant
```
Comment (scoheb) [inline demos/m4/base/release_links.yaml:9]: this should be **managed** workspace

File: demos/m4/base/release_links.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+---
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleaseLink
+metadata:
+  name: tenant
```
Comment (scoheb) [inline demos/m4/base/release_links.yaml:14]: this should be **managed** workspace

File: demos/m4/base/release_links.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  namespace: tenant
+spec:
+  displayName: Tenant's ReleaseLink
+  application: m4-app
+  target: tenant
```
Comment (scoheb) [inline demos/m4/base/release_links.yaml:19]: this should be **managed** workspace

File: demos/m4/base/release_strategy.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleaseStrategy
+metadata:
+  name: m4-strategy
+  namespace: tenant
```
Comment (scoheb) [inline demos/m4/base/release_strategy.yaml:5]: this should be **managed** workspace

File: demos/m4/base/release_strategy.yaml [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+spec:
+  pipeline: hello-world
```
Comment (scoheb) [inline demos/m4/base/release_strategy.yaml:7]: let's rename this to be m4-release-pipeline

File: demos/m4/overlays/dev/pipeline_run.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+kind: PipelineRun
+metadata:
+  name: build-m4-component
+  labels:
+    pipelines.appstudio.openshift.io/type: build
+    build.appstudio.openshift.io/component: m4-component
+  namespace: demo
+spec:
+  pipelineRef:
+    name: hello-world
```
Comment (scoheb) [inline demos/m4/overlays/dev/pipeline_run.yaml:11]: rename to be m4-release-pipeline

Comment (davidmogar) [reply demos/m4/overlays/dev/pipeline_run.yaml:11]: I'll rename it to `m4-build-pipeline` as this is not release but build.

File: demos/m4/README.md (hunk 1 above)
Comment (scoheb) [inline demos/m4/README.md:28]: still tenant

File: demos/m4/base/release_links.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  application: m4-app
+  target: managed
+---
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ReleaseLink
+metadata:
+  name: managed
+  namespace: managed
+spec:
+  displayName: Tenant's ReleaseLink
```
Comment (scoheb) [inline demos/m4/base/release_links.yaml:17]: rename to Managed Workspace's ReleaseLink

File: demos/m4/README.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
+# Milestone 4 demo
+
+This repository contains the resources needed for running the release demo for Milestone 4.
```
Comment (johnbieren) [inline demos/m4/README.md:3]: directory, not repository

**Pull request merged**